flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcServiceTest.java

```diff
@@ -361,6 +363,30 @@ public void testAkkaRpcServiceShutDownWithFailingRpcEndpoints() throws Exception
         assertThat(akkaRpcService.getActorSystem().whenTerminated().isCompleted(), is(true));
     }
 
+    @Test
+    public void failsRpcResultImmediatelyIfEndpointIsStopped() throws Exception {
```

Comment:
The code looks identical but both tests test different things: `AkkaRpcServiceTest#failsRpcResultImmediatelyIfEndpointIsStopped` tests that the `RecipientUnreachableException` is thrown if the endpoint is in the same `ActorSystem`. In contrast to that, `RemoteAkkaRpcActorTest#failsRpcResultImmediatelyIfEndpointIsStopped` tests that the same behaviour occurs if the endpoint is in a different `ActorSystem` (e.g. remote communication is needed). In fact the former case is why I had to change `AkkaInvocationHandler#resolveTimeoutException` because it only surfaced as a `AskTimeoutException` with a special message.